Updated constant velocity acquisition feature

mehr0096 opened the pull request:
This feature now acquires images at consistent intervals defined by the camera sweep time as the stage moves between user defined start and stop position at a velocity which is defined by the desired step size and camera sweep time. Camera is externally triggered by stage sync signal which is triggered when stage is at start position (scanr function moves stage beyond start position to ensure that stage is at desired velocity once stage hits start position). All move axis absolute statements were removed from feature to prevent camera acquisition from not triggering. Camera timeout was extended to 10000 seconds to allow for long finite acquisitions. External triggering was also configured in nidaq.py. Multichannel acquisition is also functional. Still need to fix jittering issue at small step sizes.

AdvancedImagingUTSW committed:
added logger and print statements

AdvancedImagingUTSW committed:
changed sweep time calculations and added log statements

AdvancedImagingUTSW committed:
changed sweep time buffer

AdvancedImagingUTSW committed:
updated scaling factor for sweep time

AdvancedImagingUTSW committed:
scaling factor

AdvancedImagingUTSW committed:
added cva_ttl and changed scaling factors

AdvancedImagingUTSW committed:
corrected encoder divide to 10 nm

AdvancedImagingUTSW committed:
changed encoder resolution to 10 and cva ttl

AdvancedImagingUTSW committed:
testing conpro projection

JinlongL committed:
Added CVA TTL feature

JinlongL committed:
changed camera channel location

JinlongL committed:
added CVA conpro and CVA ttl feature

JinlongL committed:
add different cva modes

JinlongL committed:
singlewave updates

JinlongL committed:
cva single wave updates

JinlongL committed:
changes to cvaconpro

JinlongL committed:
updated microscope galvo waveform

JinlongL committed:
saving works automatic channel selection

JinlongL committed:
added scaling factor to offset discrepancy sweeptiimes

JinlongL committed:
removed end func from cva_conpro

adjusted tolerances for cva_singlewave and cva_ttl

JinlongL committed:
added daq calls and changed stopping criteria

JinlongL committed:
implemented continuous mode

JinlongL committed:
added cva continuous mode

JinlongL committed:
adding multicolor

JinlongL committed:
multicolor testing

JinlongL committed:
implemented cva_conpro correctly by changing wait until done to 10000 seconds

need to get wait indefinitely to work

JinlongL committed:
added cva_conpro_multichannel

JinlongL committed:
fixed cva conpro start and stop added multriggering for cva_cont and continuous mode

JinlongL committed:
changed back to finite

JinlongL committed:
multichannel operational

JinlongL committed:
show positions on image writer

JinlongL committed:
Update camera_hamamatsu.py

JinlongL committed:
cleaning up

Creating a branch I don't mind destroying

Co-Authored-By: Kevin Dean <42547789+AdvancedImagingUTSW@users.noreply.github.com>

JinlongL committed:
Update constant_velocity_acquisition.py

Shit. I updated the wrong one.

Co-Authored-By: Kevin Dean <42547789+AdvancedImagingUTSW@users.noreply.github.com>

JinlongL committed:
Checkpoint

Checkpoint. Don't trust scanr or encoder counts yet..

Co-Authored-By: Kevin Dean <42547789+AdvancedImagingUTSW@users.noreply.github.com>

JinlongL committed:
Checkpoint.

Co-Authored-By: Kevin Dean <42547789+AdvancedImagingUTSW@users.noreply.github.com>

JinlongL committed:
checkpoint...

JinlongL committed:
It's functioning.

Trying to finish up the last few traceback errors. Can now switch between continuous and constant velocity without fatal errors.

JinlongL committed:
More progress.

Important things:
External trigger set to none.
Channel set to zero.
prepare acquisition
camera task retriggerable set to False
Analog tasks set to stop and close regardless of trigger mode.
camera trigger timeout...

JinlongL committed:
Notes

The commented out command for the tiger controller was also incorrect.

JinlongL committed:
updated cva conpro

JinlongL committed:
import imagewriter

JinlongL committed:
fixing cva conpro

cva conpro doesn't update to the correct number of frames

add daq trigger sent to model

JinlongL committed:
cva singlewave multicolor has issues

JinlongL committed:
change external trigger to none and encoder divide ceiling

JinlongL committed:
got cva conpro to work with external trigger

works with external trigger and works with the actual step size.

JinlongL committed:
Pushing changes

JinlongL committed:
scanv implemented in cva_cont

JinlongL committed:
implementing multicolor

JinlongL committed:
trying to implement multicolor

JinlongL committed:
multicolor cva works with external trigger on cva-ttl

cva singlewave - multicolor cva with self trigger

cva-ttl - multicolor cva with external trigger

cva-conpro-multichannel: kevin's version that works (not sure about multichannel)

cva-conpro: kevin's version for single channel

cva-cont - not sure.

JinlongL committed:
updated metadata

JinlongL committed:
fixed saving and long distance multicolor errors

JinlongL committed:
added wait until complete function for stage to prevent non triggering during acquisition

JinlongL committed:
changes to asi controller and stage buffer

JinlongL committed:
changed buffer

JinlongL committed:
fixing acquisition hanging initially and retriggering after cleanup and memory consumption after acquisition has ended

I made sure stage was moved to the correct position before the start position

I also made sure to leave update channel before new channel was prepared by getting out of update channel before channel was prepared

I also didn't set the external trigger but make the external trigger = None. This allows the next acquisition to work without retriggering.

JinlongL committed:
acquire bar and changing stop acquisition and forcing stage to be before start position

JinlongL committed:
changed stage buffer and removed some print statements

JinlongL committed:
added pausing and resuming of data threads in pre func and update channel

still need to find optimal buffer positions for stage scanning or figure out what is timing out.

JinlongL committed:
added longer wait time in run data process

Added longer wait time in run data process but what seems to happen is that the stage starts moving and hits the start position but nothing camera or voice coil doesn't turn on.

JinlongL committed:
initialized scanr for every channel

JinlongL committed:
added signal main function

JinlongL committed:
placed set speed after scanr and put scanr in the main function

I placed set speed after set speed and I placed scanr in main function. System works for 2 channels but consistently for 3rd channel.

JinlongL committed:
moved cva_ttl to cva_conpro and removved all move axis absolute in cva conpro

cva-ttl is functional multichannel external trigger. I removed all of the move axis absolute and rely on scanr to find the correct position. When the stage position is before start position to correct buffer it does send external  trigger but if the stage is too far away or after start position, system fails.

JinlongL committed:
added acquire bar to cva-conpro, conpro with external trigger seems to be functional

CVA conpro with external trigger is functional. Still need to fix saving of multiple channels (zsteps different than number of frames in metadata). Also need to look into speed across long distances.

JinlongL committed:
updated metadata configuration in save image

Updated metadata configuration in save image to make sure that each channel has same number of frames saved.

JinlongL committed:
Revert "updated metadata configuration in save image"

This reverts commit 15b3229ac4a209d7afbdf9adda0a1b7adab887cd.

JinlongL committed:
updated metadata in save image to make sure each channel has same number of frames saved

Updated metadata configuration in save image to make sure that each channel has same number of frames saved.

JinlongL committed:
Merge branch 'cva-conpro-kd-dm-v2' into cva-conpro-develop-merge

JinlongL committed:
commented out continuous acquisition

JinlongL committed:
removed alternate cva features

JinlongL committed:
removed other cva versions from acquire bar and model. Also removed configuration update in save image because it slowed down acquisition

removed other cva versions from acquire bar and model.

removed configuration update from save image because it slowed down acquisition and resulted in saving of not as many frames as intented. Also there is a class nonetype error when I did this.

JinlongL committed:
removed other cva versions from codebase

JinlongL committed:
Merge branch 'cva-conpro-develop-merge' into cva-conpro-develop-merge-v2

JinlongL committed:
removed print statements

JinlongL committed:
updated max speed calculations from max percent

AdvancedImagingUTSW commented on src/aslm/controller/sub_controllers/acquire_bar_controller.py:
```diff
@@ -80,8 +80,8 @@ def __init__(self, view, parent_view, parent_controller):
             "Confocal-Projection": "confocal-projection",
-            "ConstantVelocityAcquisition": "ConstantVelocityAcquisition",
             "Customized": "customized",
+            "CVA-Conpro":"CVACONPRO",
```
I actually prefer the feature name ConstantVelocityAcquisition

AdvancedImagingUTSW commented on src/aslm/controller/sub_controllers/acquire_bar_controller.py:
```diff
@@ -362,6 +362,11 @@ def update_stack_acq(self, mode):
             "z-stack",
             "projection",
             "ConstantVelocityAcquisition",
+            "CVATTL",
```
Most of these are unnecessary now, right?

AdvancedImagingUTSW commented on src/aslm/model/devices/APIs/asi/asi_tiger_controller.py:
```diff
@@ -466,7 +491,11 @@ def start_scan(self, axis: str, is_single_axis_scan: bool = True):
+        # Not sure if this requires an S
+        # self.send_command(f"SCAN S")
+        self.send_command(f"SCAN")
+        # self.send_command(f"SCAN S Y={fast_axis_id} Z={slow_axis_id}")
```
Remove unnecessary commented out code and comments.

AdvancedImagingUTSW commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -112,49 +113,58 @@ def set_external_trigger(self, external_trigger=None):
+            try:
+                self.camera_trigger_task.stop()
+            except:
+                print(traceback.format_exc())
```
This traceback could be turned into an f-string that is more informative. 
For example:
print(f"Error switching the camera trigger source: {traceback.format_exc()}")

AdvancedImagingUTSW commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -112,49 +113,58 @@ def set_external_trigger(self, external_trigger=None):
+                try:
+                    self.analog_output_tasks[board_name].stop()
+                except:
+                    (print(traceback.format_exc()))
```
Same here.

AdvancedImagingUTSW commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -173,11 +183,14 @@ def restart_analog_task_callback_func(task):
             Callback function
         """
 
+        # self.asi_stage = self.model.active_microscope.stages[self.axis]
```
Get rid of this line.

AdvancedImagingUTSW commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -112,49 +113,58 @@ def set_external_trigger(self, external_trigger=None):
-        )
+        self.trigger_mode = "self-trigger" if external_trigger is None else "external-trigger"
         self.external_trigger = external_trigger
+        # self.asi_stage = self.model.active_microscope.stages[self.axis]
```
Get rid of this line.

AdvancedImagingUTSW commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -173,11 +183,14 @@ def restart_analog_task_callback_func(task):
                 task.stop()
                 task.start()
-            except Exception:
+            except:
```
except Exception is better than just a bare except.

zacsimile commented on src/aslm/model/devices/APIs/hamamatsu/HamamatsuAPI.py:
```diff
@@ -765,9 +765,13 @@ def __result(self, errvalue):
         if errvalue < 0:
             try:
-                print("error message: ", DCAMERR(errvalue))
+                pass
```
This is weird. `try: pass` does nothing. Let's bring the error printing back.

AdvancedImagingUTSW replied:
How about logging it to keep the command interface uncluttered?

zacsimile commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -112,49 +113,58 @@ def set_external_trigger(self, external_trigger=None):
-        self.trigger_mode = (
-            "self-trigger" if external_trigger is None else "external-trigger"
-        )
+        self.trigger_mode = "self-trigger" if external_trigger is None else "external-trigger"
```
This edit feels like maybe the pre-commit hooks are off on your machine? Could we install them to keep the formatting consistent?

zacsimile commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -112,49 +113,58 @@ def set_external_trigger(self, external_trigger=None):
-            )
-            # change camera task to regeneratable.
-            self.camera_trigger_task.triggers.start_trigger.retriggerable = True
+                self.external_trigger)
```
Random formatting change? Why?

zacsimile commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -278,6 +294,7 @@ def create_analog_output_tasks(self, channel_key):
             self.analog_output_tasks[board].timing.cfg_samp_clk_timing(
                 rate=sample_rates[0],
                 sample_mode=nidaqmx.constants.AcquisitionType.FINITE,
+                # sample_mode=nidaqmx.constants.AcquisitionType.CONTINUOUS,
```
Delete

zacsimile commented on src/aslm/model/devices/daq/daq_ni.py:
```diff
@@ -360,37 +377,52 @@ def run_acquisition(self):
-            self.camera_trigger_task.close()
-
+        try:            
+            if self.trigger_mode == "self-trigger":
```
Why do we need these if statements? If `self.camera_trigger_task` is `None`, `self.camera_trigger_task.stop()` will throw an `AttributeError`, which we catch.

zacsimile commented on src/aslm/model/devices/remote_focus/remote_focus_base.py:
```diff
@@ -167,10 +169,7 @@ def adjust(self, exposure_times, sweep_times, offset=None):
                         zoom
                     ][channel["laser"]].get("percent_smoothing", 0.0)
                 )
-                if ps > 0:
```
Do not delete this. This is important to the program's functionality.

zacsimile commented on src/aslm/model/devices/stages/stage_asi.py:
```diff
@@ -72,47 +73,105 @@ def build_ASI_Stage_connection(com_port, baud_rate=115200):
                 axis: axes_mapping[axis] for axis in self.axes if axis in axes_mapping
-            }  #: Mapping of axes to ASI axes
-        else:
-            # Force cast axes to uppercase
```
Do not delete this. This is important to the program's functionality.

zacsimile commented on src/aslm/model/devices/stages/stage_asi.py:
```diff
@@ -72,47 +73,105 @@ def build_ASI_Stage_connection(com_port, baud_rate=115200):
-            }  #: Mapping of axes to ASI axes
-        else:
-            # Force cast axes to uppercase
-            self.axes_mapping = {k: v.upper() for k, v in self.axes_mapping.items()}
```
Do not delete this. This is important to the program's functionality.

zacsimile commented on src/aslm/model/devices/stages/stage_asi.py:
```diff
@@ -259,7 +315,7 @@ def move_axis_absolute(self, axis, abs_pos, wait_until_done=False):
             logger.exception("ASI Stage Exception", e)
             return False
-
+        
```
Remove this. Pre-commit hooks will delete these kind of things.

zacsimile commented on src/aslm/model/features/common_features.py:
```diff
@@ -1299,6 +1299,9 @@ def signal_func(self):
                 f"*** Zstack move stage: (z: {self.current_z_position}), "
                 f"(f: {self.current_focus_position})"
             )
+            
```
Unnecessary.

zacsimile commented on src/aslm/model/features/common_features.py:
```diff
@@ -1604,8 +1607,8 @@ def __init__(self, model):
         }
-
-        self.model.move_stage({"z_abs": 0})
+ 
```
Unnecessary.

zacsimile commented on src/aslm/model/features/image_writer.py:
```diff
@@ -232,6 +232,10 @@ def save_image(self, frame_ids):
 
         """
 
+        # self.data_source.set_metadata_from_configuration_experiment(
```
Unnecessary.

zacsimile commented on src/aslm/model/features/feature_container.py:
```diff
@@ -308,23 +308,20 @@ def run(self, *args, wait_response=False):
             # run(wait_response=True)
             result = self.node_funcs["main"](*args)
-
+        
```
Unnecessary.

zacsimile commented on src/aslm/model/model.py:
```diff
@@ -810,7 +807,14 @@ def run_data_process(self, num_of_frames=0, data_func=None):
                     self.logger.info("ASLM Model - Data container is closed.")
                     self.stop_acquisition = True
                     break
+                # If not constant_velocity_acquisition_mode
```
Remove

zacsimile commented on src/aslm/model/model.py:
```diff
@@ -823,12 +827,15 @@ def run_data_process(self, num_of_frames=0, data_func=None):
             self.pause_data_ready_lock.release()
 
         self.end_acquisition()  # Need this to turn off the lasers/close the shutters
+        return acquired_frame_num
```
Are you getting this returned value?

zacsimile commented:
Do we need all three constant velocity modes in here? I thought only one worked as we'd like it to.

AdvancedImagingUTSW commented on src/aslm/model/model.py:
```diff
@@ -677,7 +676,7 @@ def run_command(self, command, *args, **kwargs):
             if hasattr(self, "signal_container"):
                 self.signal_container.end_flag = True
-            if self.imaging_mode == "ConstantVelocityAcquisition":
+            if self.imaging_mode == "CVACONPRO":
```
This is a good example of intuitive naming. 

ConstantVelocityAcquisition is intuitive, and CamelCase. CVACONPRO is obscure. I get the CVA part, but the rest? Confocal projection? We aren't doing that...

AdvancedImagingUTSW commented on src/aslm/model/model.py:
```diff
@@ -810,7 +807,14 @@ def run_data_process(self, num_of_frames=0, data_func=None):
+                        # Throw away the frame.
+                        # self.number_triggers += 1
+                    # else:
+                        # Pass frame through data container.
```
Delete.

AdvancedImagingUTSW commented on src/aslm/model/devices/stages/stage_asi.py:
```diff
@@ -72,47 +73,105 @@ def build_ASI_Stage_connection(com_port, baud_rate=115200):
+        Start scan and stop after scanning
+    verify_abs_position()
+        Return a dictionary with moving positions within the min and max stage limits
+
```
We can adjust this documentation together. Sphinx doesn't recognize it in this format, which is why I went through and redid so much of the documentation.

Briefly, you don't list parameters, attributes, or methods anymore under the class section. Just title, information about the class, and perhaps a few notes.

Then, within the `__init__` section, you provide the parameters only. Sphinx automatically finds all methods.

To make Sphinx find the attributes, the line before one is defined you do something like this...

#: dict : An example test dictionary.
self.test = {}

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+        self.expand_waveform = self.model.configuration["waveform_templates"]["CVACONPRO"]["expand"]
+        print(f"repeat num = {self.repeat_waveform}")
+        print(f"expand num = {self.expand_waveform}")
+        Expand_frames = float(self.model.configuration["waveform_templates"]["CVACONPRO"]["expand"])
```
Variables names shouldn't be capitalized

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+
+
+        
+        self.asi_stage.set_speed(percent=0.0001/max_speed)
```
Why does the stage's speed get set to `0`?

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+        self.saving_flag = saving_flag
+        
+        self.image_writer = None
+        if self.saving_flag:
```
Why do we not use the standard image writer?

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+            
+            if self.current_channel_in_list>0:
+                print("PAUSING DATA THREAD IF STATEMENT IN CHANNEL")
+                self.model.pause_data_thread
```
`pause_data_thread` is a function, not an attribute. Needs to be called as `pause_data_thread()`.

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+        print(f"exposure time = {exposure_times}")
+        print(f"sweep time = {sweep_times}")
+        channel_name = next(iter(sweep_times))
+        channel_name = str(channel_name)
```
The key is a string by default.

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+        channel_name = str(channel_name)
+        channel_num = re.findall(r'[\d.]+',channel_name)
+        channel_num = int(channel_num[0])
+        self.model.active_microscope.current_channel = channel_num
```
This could be a problem. `self.model.active_microscope.current_channel` is indexing into `microscope.available_channels`. If `Channel_0` is disabled and `Channel_1` is enabled, then index `0` will point to `Channel_1`. Index `1` will point to something else.

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+        print(f"sweep time = {sweep_times}")
+        channel_name = next(iter(sweep_times))
+        channel_name = str(channel_name)
+        channel_num = re.findall(r'[\d.]+',channel_name)
```
Not wrong, but a lot of work for what could be `int(channel_name.split('_')[1])`. But also unnecessary to compute here (see comment below).

zacsimile commented on src/aslm/model/features/cva_conpro.py:
```diff
@@ -0,0 +1,484 @@
+        print("Actual Step Size of Stage:", step_size_nm)
+
+        # Calculate the actual step size in millimeters. 264 * 10^-6 mm
+        step_size_mm = step_size_nm / 1 * 10**-6  # 264 * 10^-6 mm
```
1 nm = 10^-6 mm. What is the `/ 1` for?

zacsimile commented:
I think you need to merge develop into this branch before we can merge into main.

JinlongL committed:
pre-commit hooks formatting

mehr0096 commented:
Thank you for your comments. When I merged with develop before, I resolved a few of the conflicts in the opposite direction. I will finish running pre-commit hooks, change the name of the feature, and address the other issues and comments.

AdvancedImagingUTSW commented:
Thanks, point of code review and test. Just makes everything more consistent, readable, interpretable, and functioning.

JinlongL committed:
Revert "pre-commit hooks formatting"

This reverts commit e9e79281be32680123d4016e263a30d48f02a0f1.

JinlongL committed:
changed cva_conpro to constant velocity acquisition. Removed other CVA versions and print statements and commented lines.

JinlongL committed:
Revert "changed cva_conpro to constant velocity acquisition. Removed other CVA versions and print statements and commented lines."

This reverts commit 9474129238fe674549b0b256edd3df63f7706bf8.

JinlongL committed:
changed cva_conpro to constant velocity acquisition. Removed old cva versions.

JinlongL committed:
Revert "changed cva_conpro to constant velocity acquisition. Removed old cva versions."

This reverts commit 3131f1d57ec07c663f572d4f07ec89620fa68f3a.

JinlongL committed:
changed cva conpro to constant velocity acquisition

changed cva conpro to constant velocity acquisition and fixed saving error in previous reverted commits.

JinlongL committed:
removed commented lines in daq ni and fixed pause data thread

JinlongL committed:
added self.axes_mapping and ps that was removed during merge

added axes mapping to stage asi and if ps>0 statement that was removed when merging with develop.

JinlongL committed:
updated metadata test to include CVA in stack cycling mode

JinlongL committed:
removed commented lines in model and image writer

zacsimile committed:
pre-commit hook fixes

zacsimile commented:
I ran `pre-commit run --files src/aslm/controller/sub_controllers/acquire_bar_controller.py src/aslm/model/device_startup_functions.py src/aslm/model/devices/APIs/asi/asi_tiger_controller.py src/aslm/model/devices/APIs/hamamatsu/HamamatsuAPI.py src/aslm/model/devices/camera/camera_hamamatsu.py src/aslm/model/devices/daq/daq_ni.py src/aslm/model/devices/remote_focus/remote_focus_base.py src/aslm/model/devices/stages/stage_asi.py src/aslm/model/features/common_features.py src/aslm/model/features/cva_conpro.py src/aslm/model/features/feature_container.py add src/aslm/model/metadata_sources/metadata.py src/aslm/model/microscope.py src/aslm/model/model.py test/model/metadata_sources/test_metadata.py` to clean up the code a bit before we continue reviewing.

zacsimile committed:
More cleanup

zacsimile commented:
What is the deal with `# cva_conpro_multichannel.py`? Do we delete it or keep it as a file, `cva_conpro_multichannel.py`?

zacsimile committed:
Cleanup metadata, add logging statement back

mehr0096 committed:
Delete src/aslm/model/features/# cva_conpro_multichannel.py

JinlongL committed:
removed external trigger section in stop acquisition

JinlongL committed:
changed channel number acquisition and streamlined step size nm to mm conversion.

JinlongL committed:
Revert "changed channel number acquisition and streamlined step size nm to mm conversion."

This reverts commit da96b253142f709f14fc3e62015ca075abccaaae.

JinlongL committed:
updated channel number and formatting changes

upated cva conpro to pass concurrency unit test

JinlongL committed:
cva conpro formatting changes

AdvancedImagingUTSW commented:
Is this ready for review again?

mehr0096 commented:
We can review this again.

zacsimile commented:
Re-reading the code (except for the confocal projection code) it seems more or less fine. A couple of files feel like they didn't need to be touched (e.g. `microcsope.py`), but it doesn't look like there's anything there that will break the program. I'd like to test the on a couple of other microscopes before merging to make sure the new stage code and camera task timeout aren't a problem.

JinlongL committed:
removed cva ttl from metadata

JinlongL committed:
Fix for joystick

Co-Authored-By: Zach Marin <zacsimile@users.noreply.github.com>

AdvancedImagingUTSW committed:
Convert print statements to log statements

Co-Authored-By: Zach Marin <zacsimile@users.noreply.github.com>

AdvancedImagingUTSW committed:
Murdering print statements

Also a lot of unused variables and unnecessary duplication. For example:

x = 5
self.x = x

JinlongL committed:
passing index error exception in get position in tiger controllerco

JinlongL committed:
Merge branch 'cva-conpro-develop-merge-v2-noprintstatements' of https://github.com/TheDeanLab/ASLM into cva-conpro-develop-merge-v2-noprintstatements

mehr0096 commented:
I would like to revert some of the changes made in the murdering print statements commit. I would still like to keep some of the print statements in constant_velocity_acquisition for debugging, testing, and timing purposes since there are a few bugs I still need to fix and some more features to cva I need to add. Also, I deliberately left the name of my feature cva_conpro.py so I wouldn't have to deal with merge conflicts with the existing constant velocity_acquistion,py. I was going to change the name of the feature after the merge. Let me know if this is ok.

AdvancedImagingUTSW commented:
@mehr0096 Go for it. I thought you were past the majority of the debugging phase. In the long-run, we should eliminate print statements so that our code doesn't become too cluttered. The logger should be used for routine debugging statements.

JinlongL committed:
Revert "Murdering print statements"

This reverts commit 7da7bf04e974665bb37916bf95784416f39c0a81.

JinlongL committed:
removed print statements in dummy detective, tiger controller, and conftest and added documentation to asi tiger controller

mehr0096 commented:
I added documentation to tiger controller that was removed from reverting the print statement commit. I also removed print statements from other files. I am ok with merging this. Once merged, I will change cva_conpro to constant velocity acquisition and update model.py.
